src/AutoRest.CSharp.V3/CodeGen/ClientWriter.cs

```diff
@@ -114,30 +125,32 @@ private void WriteRequestCreation(CodeWriter writer, ClientMethod operation)
                 writer.Line($"var message = pipeline.CreateMessage();");
                 writer.Line($"var request = message.Request;");
                 var method = operation.Request.Method;
-                writer.Line($"request.Method = {writer.Type(typeof(RequestMethod))}.{method.ToRequestMethodName()};");
+                writer.Line($"request.Method = {typeof(RequestMethod)}.{method.ToRequestMethodName()};");
+
+                var uriType = writer.Type(typeof(Uri));
 
                 //TODO: Add logic to escape the strings when specified, using Uri.EscapeDataString(value);
                 //TODO: Need proper logic to convert the values to strings. Right now, everything is just using default ToString().
                 //TODO: Need logic to trim duplicate slashes (/) so when combined, you don't end  up with multiple // together
                 var urlText = String.Join(String.Empty, operation.Request.HostSegments.Select(s => s.IsConstant ? s.Constant.Value : "{" + s.Parameter.Name + "}"));
-                writer.Line($"request.Uri.Reset(new {writer.Type(typeof(Uri))}($\"{urlText}\"));");
+                writer.Line($"request.Uri.Reset(new {uriType}($\"{urlText}\"));");
```

Comment:
you can use `typeof(Uri)` in format string here, `writer.Type` shouldn't be required anymore.

Reply:
Fixed.